content/hello-world-program.md

```diff
@@ -0,0 +1,295 @@
+# Hello World
+
+### List links for further reading here:
+
+- [https://docs.rs/solana-program/latest/solana_program/](https://docs.rs/solana-program/latest/solana_program/)
+
+# Objectives
+
+*By the end of this lesson, you will be able to:*
+
+- Explain the entry point to a Solana program
+
+- Submit a transaction to invoke our hello world program
+
+# TL;DR
+
+- **Accounts** on Solana each have a unique address, store data, and are owned by a program
+- **Programs** on Solana are a particular type of account marked as executable and store instruction logic
+- **Transactions** on Solana are made up of instructions which are executed by programs
+
+# Overview
+
+Solana programs are written in the Rust programming language. Solana programs are stored in accounts with a unique address referred to as the program id. To understand programs on Solana we must first understand the basics of Solana’s Account Model.
+
+### Account Model
+
+All data stored on the Solana network are contained in what are referred to as accounts. Each account has its own unique address which is used to identify and access the account data. 
+
+There are two main categories of accounts:
+
+1. Data accounts which store state (ex. name, count) and must be owned by a program account
+2. Program accounts which store the executable programs (ex. instruction logic) and process instructions 
+
+There are two types of data accounts:
+
+1. Data accounts owned by the System Program (ex. user wallets)
+2. Data accounts owned by all other programs 
+
+There are two types of program accounts:
+
+1. Native program accounts which refer to Solana’s native programs (ex. System Program)
+2. All other program accounts 
```

Comment:
@ZYJLiu **sugg:**

There are two main categories of Solana accounts:

- Data accounts
- Program accounts

Data accounts store state (e.g. name, count). Note that data accounts **_must_** be owned by a program account. There are two types of data accounts:

- Data accounts owned by the System Program (ex. user wallets)
- Data accounts owned by any other program

Program accounts store the executable programs (e.g. instruction logic) and process instructions . There are two types of program accounts:

- Native program accounts which refer to Solana’s native programs (e.g. System Program)
- All other program accounts